src/exeptions/api.error.js

```diff
@@ -0,0 +1,33 @@
+class ApiError extends Error {
+  constructor({ message, status, errors = {} }) {
+    super(message);
+
+    this.status = status;
+    this.errors = errors;
+  }
+
+  static badRequest(message, errors) {
+    return new ApiError({
+      message,
+      errors,
+      status: 400,
+    });
+  }
+
+  static unauthorized(errors) {
+    return new ApiError({
+      message: 'unauthorized user',
+      errors,
+      status: 404,
```

Comment:
The status code for `unauthorized` should be 401, not 404. Consider changing `status: 404` to `status: 401`.